sound/soc/soc-dapm.c

```diff
@@ -916,6 +916,8 @@ static int dapm_create_or_share_kcontrol(struct snd_soc_dapm_widget *w,
 				return -EINVAL;
 			}
 		}
+		if (w->no_wname_in_long_name)
+			wname_in_long_name = false;
```

Comment:
Writing technical documentation is difficult... 
Can I suggest the following commit message:

"
The existing soc-dapm code may add a prefix to control names, which in some cases is useful but in others leads to long and confusing control names such as  "gain 2.1 Main Playback Volume".

This patch suggests an added flag to prevent the widget name prefix from being added. That flag will be set in the topology file on a per-widget basis
"